src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserStateObjectFactory.Windows.cs

```diff
@@ -17,9 +17,15 @@ internal sealed class TdsParserStateObjectFactory
         // Temporary disabling App Context switching for managed SNI.
         // If the appcontext switch is set then Use Managed SNI based on the value. Otherwise Managed SNI should always be used.
         //private static bool shouldUseLegacyNetorking;
+        private static Lazy<bool> shouldUseLegacyNetorking = new Lazy<bool>(() => bool.TrueString.Equals(Environment.GetEnvironmentVariable("System.Data.SqlClient.UseLegacyNetworkingOnWindows"), StringComparison.InvariantCultureIgnoreCase));
```

Comment:
This line can be in the #DEBUG condition as well.

Reply:
done.